dev/com.ibm.ws.logging_fat/fat/src/com/ibm/ws/logging/fat/StackTraceJoinerTest.java

```diff
@@ -117,7 +120,39 @@ public void testPrintStackTraceOverrideReflection() throws Exception {
     }
 
     @Test
-    public void testPrintStackTraceMultipleThreads() {
-        // TODO: Test printStackTrace in multiple threads and ensure that messages are not leaking from one stack trace to another
+    public void testPrintStackTraceMultipleThreads() throws Exception {
+        final int threadsToRun = 3;
+        final int expectedErrOut = threadsToRun + 3;
+        Logger logger = Logger.getLogger("com.ibm.ws.logging.stackjoiner");
+        logger.log(Level.INFO, ">>>>> Create threads");
+
+        for (int i = 0; i < threadsToRun; i++) {
+            new Thread(new Runnable() {
+                @Override
+                public void run() {
+                    try {
+                      messagesLog = server.getDefaultLogFile();
+                      server.setMarkToEndOfLog(messagesLog);
+                      hitWebPage("broken-servlet", "ExceptionPrintingServlet", false);
+
+                    } catch (Exception e) {
+                        // TODO Auto-generated catch block
+                        e.printStackTrace();
+                    }
+                }
+            }).start();
+            logger.log(Level.INFO, "Thread-" + (i + 1) + " started.");
+            try {
+              Thread.sleep(500);
```

Comment:
you'd be better off keeping a reference to each of the threads you start and using Thread.join to wait for each of the threads to complete.